test/python/qasm2/test_circuit_methods.py

```diff
@@ -293,7 +298,7 @@ def test_from_qasm_str_custom_gate1(self):
         expected = QuantumCircuit(qr, name="circuit")
         expected.append(rinv, [qr[0]])
 
-        self.assertEqualUnroll(["sdg", "h"], circuit, expected)
+        self.assertEqualUnroll(["sdg", "h", "rinv"], circuit, expected)
```

Comment:
I'm a bit confused about these changes. Wouldn't it be sufficient to change how `assertEqualUnroll` works? It seems here that the basis translator will simply not unroll the circuit anymore, would it?

Reply:
At first, I tried to only change how assertEqualUnroll works, but the transition from using `Unroller` to `BasisTranslator` was not as smooth as I thought. The basis translaator will not unroll many of the circuits with custom gates, and if we do not include the custom gate in the basis like `self.assertEqualUnroll(["sdg", "h"], circuit, expected)`, then the output will be 
`qiskit.transpiler.exceptions.TranspilerError: "Unable to translate the operations in the circuit: ['rinv'] to the backend's (or manually specified) target basis: ['snapshot', 'delay', 'h', 'measure', 'reset', 'sdg', 'barrier']. This likely means the target basis is not universal or there are additional equivalence rules needed in the EquivalenceLibrary being used. For more details on this error see: https://docs.quantum-computing.ibm.com/api/qiskit/qiskit.transpiler.passes.BasisTranslator#translation-errors"`